File: lib-base/src/main/java/com/llj/lib/base/ICommon.java
package com.llj.lib.base;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.view.View;

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/24
 */
public interface ICommon {
    default void getIntentData(Intent intent) {
    }

    default View layoutView() {
        return null;
    }

    @LayoutRes
    int layoutId();

    void initViews(Bundle savedInstanceState);

    void initData();
}

File: lib-base/src/main/java/com/llj/lib/base/ICommon.kt
package com.llj.lib.base

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.annotation.LayoutRes
import androidx.viewbinding.ViewBinding

/**
 * ArchitectureDemo
 * describe:通用的方法
 * @author llj
 * @date 2018/5/24
 */
interface ICommon<V : ViewBinding> : IWindowInset, IReflectView<V> {

  fun getIntentData(intent: Intent) {}


  fun getArgumentsData(bundle: Bundle?) {}

  //返回Boolean说明是否预加载数据
  fun preLoadData(): Boolean {
    return false
  }

  fun layoutView(): View? {
    return null
  }

  @LayoutRes
  fun layoutId(): Int

  fun initViews(savedInstanceState: Bundle?)

  fun initData()
}
